tests/db/test_bigchain_api.py

```diff
@@ -232,236 +99,22 @@ def test_text_search(self, b, alice):
-        from bigchaindb.backend import query
-        # get the number of blocks
-        num_blocks = query.count_blocks(b.connection)
-
-        # get the last block
-        last_block = b.get_last_block()
-
-        assert last_block['block']['block_number'] == num_blocks - 1
-
-    @pytest.mark.skipif(reason='This test may not make sense after changing the chainification mode')
-    def test_get_last_block_id(self, b):
-        last_block = b.get_last_block()
-        last_block_id = b.get_last_block_id()
-
-        assert last_block_id == last_block['id']
-
-    @pytest.mark.skipif(reason='This test may not make sense after changing the chainification mode')
-    def test_get_previous_block(self, b):
-        last_block = b.get_last_block()
-        new_block = b.create_block([])
-        b.write_block(new_block)
-
-        prev_block = b.get_previous_block(new_block)
-
-        assert prev_block == last_block
-
-    @pytest.mark.skipif(reason='This test may not make sense after changing the chainification mode')
-    def test_get_previous_block_id(self, b):
-        last_block = b.get_last_block()
-        new_block = b.create_block([])
-        b.write_block(new_block)
-
-        prev_block_id = b.get_previous_block_id(new_block)
-
-        assert prev_block_id == last_block['id']
-
-    def test_create_empty_block(self, b):
-        from bigchaindb.common.exceptions import OperationError
-
-        with pytest.raises(OperationError) as excinfo:
-            b.create_block([])
-
-        assert excinfo.value.args[0] == 'Empty block creation is not allowed'
-
     @pytest.mark.usefixtures('inputs')
+    @pytest.mark.tendermint
     def test_non_create_input_not_found(self, b, user_pk):
         from cryptoconditions import Ed25519Sha256
         from bigchaindb.common.exceptions import InputDoesNotExist
         from bigchaindb.common.transaction import Input, TransactionLink
         from bigchaindb.models import Transaction
-        from bigchaindb import BigchainDB
 
         # Create an input for a non existing transaction
         input = Input(Ed25519Sha256(public_key=b58decode(user_pk)),
                       [user_pk],
                       TransactionLink('somethingsomething', 0))
         tx = Transaction.transfer([input], [([user_pk], 1)],
                                   asset_id='mock_asset_link')
-
         with pytest.raises(InputDoesNotExist):
-            tx.validate(BigchainDB())
-
-    def test_count_backlog(self, b, user_pk, alice):
-        from bigchaindb.backend import query
-        from bigchaindb.models import Transaction
-
-        for i in range(4):
-            tx = Transaction.create([alice.public_key], [([user_pk], 1)],
-                                    metadata={'msg': i}) \
-                            .sign([alice.private_key])
-            b.write_transaction(tx)
-
-        assert query.count_backlog(b.connection) == 4
+            tx.validate(b)
 
 
 class TestTransactionValidation(object):
```

Comment:
There are tests still disabled starting from this one. Do you plan to enable them in a separate PR?

Reply:
Yes. I'll keep working on this until I've done all the tests in this file, but this seemed like a natural break point where I could commit some functionality.

Reply:
I think we should merge this one and make a part two PR instead